### ISSUE TITLE
1.4.1 symbols are not in Microsoft's Symbol Server

### ISSUE THREAD

**Issue by brianqcgames** [dotnet/orleans#2497]
After updating to 1.3.1 I cannot load the symbols for any orleans library.

Here is the symbol load information
C:\SymbolCache\Orleans.pdb\cf67e00b7fbb46a9abcbfa9e7f702b981\Orleans.pdb: Cannot find or open the PDB file.
C:\SymbolCache\MicrosoftPublicSymbols\Orleans.pdb\cf67e00b7fbb46a9abcbfa9e7f702b981\Orleans.pdb: Cannot find or open the PDB file.
https://msdl.microsoft.com/download/symbols: Skipped checking location. The symbol file was not found in this location before.

**Comment by sergeybykov** [comment]
I checked - I'm able to load the 1.3.1 symbols with the checked "Microsoft Symbol Server" option in VS 2015 when doing this from home. So it should work for everyone. @brianqcgames, is it still not working for you?

**Comment by brianqcgames** [comment]
Interestingly enough, I didn't change anything but just noticed symbols are loading properly this morning.

Unless anyone else is seeing this, we can close this issue.

**Comment by peter-perot** [comment]
@sergeybykov I updated from Orleans 1.2.3 to 1.4.1 and wanted to test debugging into Orleans source code using the symbol server, but it does not work. Orleans symbols are not loaded. This are the only symbols in my symbol cache:

```
02.05.2017  22:43    <DIR>          .
02.05.2017  22:43    <DIR>          ..
02.05.2017  22:43    <DIR>          Accessibility.pdb
02.05.2017  22:31    <DIR>          FailedLoads
02.05.2017  22:19    <DIR>          Microsoft.CodeAnalysis.CSharp.pdb
02.05.2017  22:19    <DIR>          Microsoft.CodeAnalysis.pdb
02.05.2017  22:19    <DIR>          Microsoft.Extensions.DependencyInjection.Abstractions.pdb
02.05.2017  22:19    <DIR>          Microsoft.Extensions.DependencyInjection.pdb
02.05.2017  22:42    <DIR>          Microsoft.VisualStudio.HostingProcess.Utilities.pdb
02.05.2017  22:42    <DIR>          Microsoft.VisualStudio.HostingProcess.Utilities.Sync.pdb
02.05.2017  22:30    <DIR>          MicrosoftPublicSymbols
02.05.2017  22:19    <DIR>          mscorlib.pdb
02.05.2017  22:19                 0 pingme.txt
02.05.2017  22:19    <DIR>          System.Collections.Immutable.pdb
02.05.2017  22:19    <DIR>          System.ComponentModel.Composition.pdb
02.05.2017  22:20    <DIR>          System.ComponentModel.DataAnnotations.pdb
02.05.2017  22:20    <DIR>          System.Configuration.Install.pdb
02.05.2017  22:19    <DIR>          System.Configuration.pdb
02.05.2017  22:19    <DIR>          System.Core.pdb
02.05.2017  22:20    <DIR>          System.Data.Entity.pdb
02.05.2017  22:20    <DIR>          System.Data.pdb
02.05.2017  22:20    <DIR>          System.Data.SqlXml.pdb
02.05.2017  22:20    <DIR>          System.DirectoryServices.Protocols.pdb
02.05.2017  22:20    <DIR>          System.Drawing.pdb
02.05.2017  22:19    <DIR>          System.IO.Compression.pdb
02.05.2017  22:19    <DIR>          System.IO.FileSystem.pdb
02.05.2017  22:20    <DIR>          System.Management.pdb
02.05.2017  22:19    <DIR>          System.Numerics.pdb
02.05.2017  22:19    <DIR>          System.pdb
02.05.2017  22:19    <DIR>          System.Reflection.Metadata.pdb
02.05.2017  22:19    <DIR>          System.Runtime.Serialization.pdb
02.05.2017  22:20    <DIR>          System.Security.Cryptography.Algorithms.pdb
02.05.2017  22:20    <DIR>          System.Security.Cryptography.Primitives.pdb
02.05.2017  22:20    <DIR>          System.Security.pdb
02.05.2017  22:19    <DIR>          System.ServiceModel.Internals.pdb
02.05.2017  22:20    <DIR>          System.Transactions.pdb
02.05.2017  22:19    <DIR>          System.ValueTuple.pdb
02.05.2017  22:42    <DIR>          System.Windows.Forms.pdb
02.05.2017  22:19    <DIR>          System.Xml.Linq.pdb
02.05.2017  22:19    <DIR>          System.Xml.pdb
02.05.2017  22:42    <DIR>          vshost32.pdb
```

I checked the "Microsoft Symbol Servers" checkbox in VS 2015, but it does not help.

**Comment by sergeybykov** [comment]
Looks like symbols for 1.4.1 are indeed not on the symbol servers.

**Comment by sergeybykov** [comment]
I resubmitted 1.4.1 symbols for indexing. It may take up to 2 days for them to go through the pipeline, if everything goes fine.

**Comment by sergeybykov** [comment]
Symbols appear to be live on public symbol servers now.

**Comment by sergeybykov** [comment]
@peter-perot Thanks for reporting this! Can you verify that it's working for you now?

**Comment by peter-perot** [comment]
@sergeybykov Symbols are now loaded into the cache directory, but source code does not open. I get the following message when trying to debug a `TypeLoadException` which is originating from Orleans code:

`Cannot find file C:\Builds\8321\Orleans\1.4.1-Build-Only\src\src\Orleans\CodeGeneration\TypeUtils.cs`

When I click `Cancel`another source code file is reported missing and so on.

**Comment by sergeybykov** [comment]
Do you have sources of 1.4.1 (https://github.com/dotnet/orleans/releases/tag/v1.4.1) checked out? If so, you should be able to point debugger to the source file from there. Does this not work?

**Comment by peter-perot** [comment]
@sergeybykov Yes, it works when downloading the sources, but breakpoints I set into Orleans code are not hit. I'm trying to figure out why Orleans is throwing a `TypeLoadException` on a specific grain, but currently I have no idea what's happening.

**Comment by sergeybykov** [comment]
Do you mean the debugger matches the sources to the symbols fine, but your breakpoint isn't getting hit anyway? What's the call stack of `TypeLoadException`?

**Comment by peter-perot** [comment]
The exception occurs in `TypeUtil.cs` in method `GetDefinedTypes()` when accessing the property `assembly.DefinedTypes`. When I set a breakpoint at the start of the method, it is not hit.

I do not know how to track down why a specific type cannot be loaded - the exception is pretty unspecific.

Some source code files do not seem to match the symbols, e.g. when stepping through `Silo.cs` the debugger is completely out of sync.

**Comment by sergeybykov** [comment]
>I do not know how to track down why a specific type cannot be loaded - the exception is pretty unspecific.

I suspect this may be coming from the .NET type loading and resolution. Try referencing a type from the assembly that is failing to load explicitly, e.g. with `var t = typeof(<type from that assembly>)` to force loading of it. Tagging @ReubenBond because he's dealt with such issues quite a bit.

>Some source code files do not seem to match the symbols, e.g. when stepping through Silo.cs the debugger is completely out of sync.

I think this is because of the CR/LF discrepancy. If I download https://github.com/dotnet/orleans/archive/v1.4.1.zip, debugger complains that the sources don't match. If I check out the same v1.4.1 tag locally, everything matches fine, presumably because git converts CR/LFs automatically for Windows. Did you download sources from GitHub of checked out the branch?

**Comment by peter-perot** [comment]
@sergeybykov Thank you, your workaround works!

There is another issue I opened some time ago ([Issue 1480](https://github.com/dotnet/orleans/issues/1480#issuecomment-238844791)) where I implemented an `EagerLoader` in order to eagerly load all assemblies which

* are not Orleans assemblies
* but are assemblies which reference an Orleans assembly (any assembly the name of which is starting with "Orleans").

I used this `EagerLoader` on grain services but now I have a project which is a grain client and also runs into these reflection issues. However the `EagerLoader` also works client side, but it seems still to be an unresolved issue which @ReubenBond is working on.

**Comment by peter-perot** [comment]
@sergeybykov wrote:

> Did you download sources from GitHub of checked out the branch?

I downloaded the ZIP file from github so this actually seems to be a CR/LF issue.

**Comment by sergeybykov** [comment]
For 2.0 we are considering switching to more explicit loading of application assemblies as part of `SiloBuilder`. We hope that among other things that will help with such type loading issues.

**Comment by ibebbs** [comment]
@sergeybykov , I don't seem to be able to load symbols for 1.5.0 from the Microsoft Symbol Servers. Is this just because it "may take up to 2 days for them to go through the pipeline" or have they not been submitted?

Thanks.

**Comment by sergeybykov** [comment]
1.5.0 symbols were successfully submitted to the publishing pipeline, but I still don't see them either (I happened to check a few minutes ago). I was going to wait for another day before starting inquiring.

**Comment by sergeybykov** [comment]
I just checked, and 1.5.0 symbols loaded from the symbol server fine for me.
